src/nupic/regions/BacktrackingTMRegion.cpp

```diff
@@ -0,0 +1,1191 @@
+    Size size = args_.numberOfCols * args_.cellsPerColumn;
+    getOutput("lrnActiveStateT")->getData() = Array(NTA_BasicType_Byte, lrn, size);
+
+    auto activeColumns = VectorHelpers::binaryToSparse(bottomUpIn.asVector<Real32>());
+    Real32 anomalyScore = algorithms::anomaly::computeRawAnomalyScore(
+        activeColumns, prevPredictedColumns_);
+    getOutput("anomalyScore")->getData() = Array(NTA_BasicType_Real32, &anomalyScore, 1);
+  }
+
+  if (args_.computePredictedActiveCellIndices) {
+    Output *activeCells = getOutput("activeCells");
+    Output *predictedActiveCells = getOutput("predictedActiveCells");
+    Byte *activeState = tm_->getActiveState();
+    Size nCells = args_.numberOfCols * args_.cellsPerColumn;
+    NTA_ASSERT(activeCells != nullptr);
+    NTA_ASSERT(predictedActiveCells != nullptr);
+    NTA_ASSERT(args_.outputWidth == nCells);
+    NTA_ASSERT(args_.outputWidth == activeCells->getData().getCount());
+    NTA_ASSERT(args_.outputWidth == predictedActiveCells->getData().getCount());
+
+    Real32 *activeCellsPtr = (Real32 *)activeCells->getData().getBuffer();
+    Real32 *predictedActiveCellsPtr =  (Real32 *)predictedActiveCells->getData().getBuffer();
+    for (size_t idx = 0; idx < nCells; idx++) {
+      activeCellsPtr[idx] = (activeState[idx]) ? 1.0f : 0.0f;
+      predictedActiveCellsPtr[idx] = (prevPredictedState_[idx] && activeState[idx]) ? 1.0f : 0.0f;
+    }
+  }
+}
+
+std::string BacktrackingTMRegion::executeCommand(const std::vector<std::string> &args,
+                                     Int64 index) {
+  // The TM does not execute any Commands.
+  return "";
+}
+
+// This is the per-node output size. This determines how big the output
+// buffers should be allocated to during Region::initialization(). NOTE: Some
+// outputs are optional, return 0 if not used.
+size_t BacktrackingTMRegion::getNodeOutputElementCount(const std::string &outputName) const {
+  if (outputName == "bottomUpOut")
+    return (args_.orColumnOutputs)? args_.numberOfCols: args_.outputWidth;
+  if (outputName == "topDownOut")
+    return args_.numberOfCols;
+  if (outputName == "lrnActiveStateT")
+    return args_.outputWidth;
+  if (outputName == "activeCells")
+    return args_.outputWidth;
+  if (outputName == "predictedActiveCells")
+    return args_.outputWidth;
+  return 0; // an optional output that we don't use.
+}
+// Note: - this is called during Region initialization, after configuration
+//         is set but prior to calling initialize on this class to create the tm. 
+//         The input dimensions should already have been set, normally from 
+//         its connected output. This would set the region dimensions if not overridden.
+//       - This is not called if output dimensions were explicitly set for this output.
+//       - This call determines the dimensions set on the Output buffers.
+Dimensions BacktrackingTMRegion::askImplForOutputDimensions(const std::string &name) {
```

Comment:
> I think I've seen this elsewhere, can be moved to RegionImpl to avoid duplicity?

It IS in RegionImpl.  

The region Impl has the right to override the default behavior if it needs to do something a little different.  In this case it needs to check the parameter 'orColumnOutputs' to decide what the output dimensions should be....if it should leave room for the cells.